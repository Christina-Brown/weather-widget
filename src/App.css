* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.app {
  background-image: url("./assets/sunny.jpg");
  background-size: cover;
  background-position: bottom;
  transition: 0.5 ease-out;
}

.app-cloudy {
  background-image: url("./assets/cloudy.jpg");
}

.app-sunny {
  background-image: url("./assets/sunny.jpg");
}

.app-sunset {
  background-image: url("./assets/sunset.jpg");
}

.app-dawn {
  background-image: url("./assets/dawn.jpg");
}

.app-night {
  background-image: url("./assets/night.jpg");
  background-color: pink;
}

main {
  min-height: 100vh;
  background-image: linear-gradient(
    to bottom,
    rgba(0, 0, 255, 0.2),
    rgba(0, 0, 0, 0.5)
  );
  padding: 25px;
  color: #fff;
}

.weather-image {
  width: 15vw;
}
