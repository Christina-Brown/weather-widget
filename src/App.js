import React, { useEffect, useState } from "react";

import { getApi } from "./utils.js";
import CityInfo from "./components/cityInfo/cityInfo";
import Forecast from "./components/forecast/forecast";
import Search from "./components/search/search";
import "./App.css";

function App() {
  const [searchString, setSearchString] = useState("");
  const [weatherData, setWeatherData] = useState({});
  const [forecast, setForecast] = useState({});

  useEffect(() => {
    if (!searchString) {
      navigator.geolocation.getCurrentPosition(
        geoLocationWeather,
        geoLocationError
      );
    }
  }, []);

  useEffect(() => {
    if (weatherData && weatherData.coord) {
      getForecast();
    }
  }, [weatherData]);

  function onSearch(e) {
    if (e.key.toLowerCase() === "enter" && searchString) {
      fetch(
        `${getApi.base}weather?q=${searchString}&units=metric&appid=${getApi.key}`
      )
        .then((response) => response.json())
        .then((jsonResponse) => {
          setWeatherData(jsonResponse);
        });
    }
  }

  function geoLocationWeather(location) {
    let longitude = location.coords.longitude;
    let latitude = location.coords.latitude;
    fetch(
      `${getApi.base}weather?lat=${latitude}&lon=${longitude}&units=metric&appid=${getApi.key}`
    )
      .then((response) => response.json())
      .then((jsonResponse) => {
        setWeatherData(jsonResponse);
      });
  }

  function geoLocationError() {
    let location = {
      coords: {
        longitude: "-0.13",
        latitude: "51.51",
      },
    };
    geoLocationWeather(location);
  }

  function getForecast() {
    let longitude = weatherData.coord.lon;
    let latitude = weatherData.coord.lat;

    fetch(
      `${getApi.base}onecall?lat=${latitude}&lon=${longitude}&exclude=hourly&units=metric&appid=${getApi.key}`
    )
      .then((response) => response.json())
      .then((jsonResponse) => {
        setForecast(jsonResponse);
      });
  }

  return (
    <div className="app">
      <main>
        <Search
          onChange={(e) => setSearchString(e.target.value)}
          onKeyUp={onSearch}
        />
        {weatherData.main ? (
          <div>
            <CityInfo weatherData={weatherData} />
            {forecast?.daily && <Forecast dailyForecast={forecast.daily} />}
          </div>
        ) : (
          <div>Loading...</div>
        )}
      </main>
    </div>
  );
}

export default App;
