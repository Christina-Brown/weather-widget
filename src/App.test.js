import React from "react";
import { mount } from "enzyme";
import App from "./App";

let wrapper;

beforeEach(() => {
  const mockGeolocation = {
    getCurrentPosition: jest.fn().mockImplementationOnce(() =>
      Promise.resolve({
        coords: {
          latitude: 51.1,
          longitude: 45.3,
        },
      })
    ),
  };
  global.navigator.geolocation = mockGeolocation;
  wrapper = mount(<App />);
});

it("should render App component without crashing", () => {
  expect(wrapper).toMatchSnapshot();
});
