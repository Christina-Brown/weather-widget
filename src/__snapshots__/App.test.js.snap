// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component without crashing 1`] = `
<App>
  <div
    className="app"
  >
    <main>
      <Search
        onChange={[Function]}
        onKeyUp={[Function]}
      >
        <div
          className="search"
        >
          <input
            className="search-bar"
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Search a city..."
            type="text"
          />
        </div>
      </Search>
      <div>
        Loading...
      </div>
    </main>
  </div>
</App>
`;
